Scheduled message notification flash

zaksoup opened the pull request:
Still missing:
* link to scheduled message list
* editing scheduled messages

tmaybe commented:
also missing-scheduled messages alert as described in #48; it's an element distinct from the flash.

tmaybe commented:
...and should be implemented before the list and editing

zaksoup commented:
I'm not super happy with the change I just pushed but it gets us out the door. I think we should pull some logic out of the messages controller and into a library -- that way we can test the business logic around scheduled vs. immediate messages better.

Commit:
Show notification when user schedules a text message

* incomplete test for this functionality

Signed-off-by: Zak Auerbach <zauerbach@codeforamerica.org>
Signed-off-by: Tomas Apodaca <tomas@codeforamerica.org>

Commit:
Do not display message in timeline until it has been sent to twilio

Signed-off-by: Tomas Apodaca <tomas@codeforamerica.org>

Commit:
Remove is_update from message broadcast

* This is unnecessary - the front-end should determine how to update

Signed-off-by: Paras Sanghavi <paras@codeforamerica.org>
Signed-off-by: Tomas Apodaca <tomas@codeforamerica.org>

Commit:
first pass at scheduled message link

CSS is not production ready.
Also moved new message into view from controller.

Signed-off-by: Zak Auerbach <zauerbach@codeforamerica.org>
Signed-off-by: Tomas Apodaca <tomas@codeforamerica.org>

Commit:
Better code quality

* prefer === to ==
* shorter create controller method

Signed-off-by: Tomas Apodaca <tomas@codeforamerica.org>

Commit:
failing test for scheduled messages link

Commit:
Force 2-digit hrs/mins so capybara can match

* The hours/mins selects have two digit numbers (.e.g. 06). Capybara
* was trying to match a single digit (e.g. 6) and finding all select options with
* that digit.

Signed-off-by: Tomas Apodaca <tomas@codeforamerica.org>

Commit:
Do not show messages until they've been sent when reloading

Signed-off-by: Zak Auerbach <zauerbach@codeforamerica.org>

Commit:
Remove scheduled message link

* add in scheduled_messages helper and test
* remove dummy from scheduled_messages helper spec, it has a helper
helper

Commit:
Refactor removing scheduled_messages link

Signed-off-by: Zak Auerbach <zauerbach@codeforamerica.org>

Commit:
Add back test for schedule message job

*accidentally deleted instead of scheduled_message_broadcast_job

Commit:
move scheduled message link into a partial
start on scheduled message broadcast job
failing test for live update of scheduled message link

Commit:
Broadcast updated scheduled message link when message is sent

* also show a flash notice when scheduling a message

Signed-off-by: Zak Auerbach <zauerbach@codeforamerica.org>

Commit:
Add back in accidentally removed test logic

Signed-off-by: Zak Auerbach <zauerbach@codeforamerica.org>

Commit:
Add jasmine to gemfile

Signed-off-by: Tomas Apodaca <tomas@codeforamerica.org>
